shared/pkg/analyzer/trivy/trivy.go

```diff
@@ -0,0 +1,125 @@
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trivy
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/aquasecurity/trivy/pkg/commands/artifact"
+	trivyFlag "github.com/aquasecurity/trivy/pkg/flag"
+
+	"github.com/openclarity/kubeclarity/shared/pkg/analyzer"
+	"github.com/openclarity/kubeclarity/shared/pkg/config"
+	"github.com/openclarity/kubeclarity/shared/pkg/formatter"
+	"github.com/openclarity/kubeclarity/shared/pkg/job_manager"
+	"github.com/openclarity/kubeclarity/shared/pkg/utils"
+)
+
+const AnalyzerName = "trivy"
+
+type Analyzer struct {
+	name       string
+	logger     *log.Entry
+	config     config.AnalyzerTrivyConfigEx
+	resultChan chan job_manager.Result
+	localImage bool
+}
+
+func New(conf *config.Config,
+	logger *log.Entry,
+	resultChan chan job_manager.Result,
+) job_manager.Job {
+	return &Analyzer{
+		name:       AnalyzerName,
+		logger:     logger.Dup().WithField("analyzer", AnalyzerName),
+		config:     config.CreateAnalyzerTrivyConfigEx(conf.Analyzer, conf.Registry),
+		resultChan: resultChan,
+		localImage: conf.LocalImageScan,
+	}
+}
+
+func (a *Analyzer) Run(sourceType utils.SourceType, userInput string) error {
+	a.logger.Infof("Called %s analyzer on source %v %v", a.name, sourceType, userInput)
+	go func() {
+		res := &analyzer.Results{}
+
+		var output bytes.Buffer
+		trivyOptions := trivyFlag.Options{
+			GlobalOptions: trivyFlag.GlobalOptions{
+				Timeout: a.config.Timeout,
+			},
+			ScanOptions: trivyFlag.ScanOptions{
+				Target:         userInput,
+				SecurityChecks: nil, // Disable all security checks for SBOM only scan
+			},
+			ReportOptions: trivyFlag.ReportOptions{
+				Format:       "cyclonedx", // Cyconedx format for SBOM so that we don't need to convert
+				ReportFormat: "all",       // Full report not just summary
+				Output:       &output,     // Save the output to our local buffer instead of Stdout
+				ListAllPkgs:  true,        // List all packages for full SBOM
+			},
+		}
+
+		var trivySourceType artifact.TargetKind
+		switch sourceType {
+		case utils.IMAGE:
+			trivySourceType = artifact.TargetContainerImage
+		case utils.ROOTFS:
+			trivySourceType = artifact.TargetRootfs
+		case utils.DIR, utils.FILE:
+			trivySourceType = artifact.TargetFilesystem
+		case utils.SBOM:
+			fallthrough
+		default:
+			a.logger.Infof("Skipping analyze unsupported source type: %s", sourceType)
+			a.resultChan <- res
+			return
+		}
+
+		err := artifact.Run(context.TODO(), trivyOptions, trivySourceType)
+		if err != nil {
+			a.setError(res, fmt.Errorf("failed to generate SBOM: %w", err))
+			return
+		}
+
+		frm := formatter.New(formatter.CycloneDXJSONFormat, output.Bytes())
```

Comment:
shouldn't this be CycloneDX XML? i see that the merge sbom logic doing formatting, but maybe better to have a uniform return format?